### PR TITLE
Order preserving fixes and remove unessesary resources

### PR DESCRIPTION
**Pull request by Yasara123**
Modify thread pool configurations in source. 
Add order preserve functionality.
Remove unnecessary util classes.
Add more description.
Modify exceptions with throw-able.
Add more test cases.
Remove executor service in sink

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 1]
```diff
@@ -48,16 +49,16 @@
...
-import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 
 /**
  * {@code HttpSink } Handle the HTTP publishing tasks.
  */
 @Extension(name = "http", namespace = "sink",
-        description = "This extension handle the output transport via http using carbon transport ", parameters = {
+        description = "This is description for http sink extension. This extension handle the output transport via " +
+                "http or https protocols using carbon transport. As the additional features this component can " +
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: For the end user carbon transport is irrelevant, explain what this do.

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 2]
```diff
@@ -74,10 +75,12 @@
...
                 " to be send data. If basic auth enable then this is a mandatory argument", type =
                 {DataType.STRING}),
-        @Parameter(name = "client.truststore.path", description = "user can give custom client trusore if user never " +
-                "mention such then system use default client-trustore in ${carbon.home}/conf/security folder", type =
+        @Parameter(name = "client.truststore.path", description = "user can give custom client trusts-sore if user " +
+                "never " +
+                "mention such then system use default client-trusts-sore in ${carbon.home}/conf/security folder", type =
                 {DataType.STRING}),
-        @Parameter(name = "client.truststore.pass", description = "user can give custom client trusore pass if user " +
+        @Parameter(name = "client.truststore.pass", description = "user can give custom client trusts-sore pass if " +
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: trusts-sore typo

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 3]
```diff
@@ -74,10 +75,12 @@
         @Parameter(name = "basic.auth.password", description = "by this user can enable give their password of server" +
                 " to be send data. If basic auth enable then this is a mandatory argument", type =
                 {DataType.STRING}),
-        @Parameter(name = "client.truststore.path", description = "user can give custom client trusore if user never " +
-                "mention such then system use default client-trustore in ${carbon.home}/conf/security folder", type =
+        @Parameter(name = "client.truststore.path", description = "user can give custom client trusts-sore if user " +
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: trusts-sore typo

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 4]
```diff
@@ -131,13 +134,13 @@
                 ),
                 @SystemParameter(
                         name = "https.trustStoreFile",
-                        description = "The default truststore file path.",
+                        description = "The default trusts-store file path.",
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: why this renaming ?

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 5]
```diff
@@ -19,6 +19,10 @@
 
 package org.wso2.extension.siddhi.io.http.source;
 
+
+
+
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: why new lines ?

File: component/src/main/java/org/wso2/extension/siddhi/io/http/util/HttpConstants.java [hunk 6]
```diff
@@ -38,34 +73,18 @@ private HttpConstants() {
     public static final String IS_AUTHENTICATION_REQUIRED = "basic.auth.enabled";
     public static final String RECEIVER_USERNAME = "basic.auth.username";
     public static final String RECEIVER_PASSWORD = "basic.auth.password";
-    public static final String CLIENT_TRUSTSTORE_PATH = "client.truststore.path";
+    public static final String CLIENT_TRUSTSTORE_PATH = "client.trustsstore.path";
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/util/HttpConstants.java]: i think its truststore !

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java (hunk 4 above)
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: why no dots used as : https.truststore.file ?

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java (hunk 1 above)
Comment (Yasara123) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: Fixed in  	c117e5f

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java (hunk 3 above)
Comment (Yasara123) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: Fixed in c117e5f

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java (hunk 2 above)
Comment (Yasara123) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: Fixed in c117e5f

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java (hunk 5 above)
Comment (Yasara123) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: Fixed in c117e5f

File: component/src/main/java/org/wso2/extension/siddhi/io/http/util/HttpConstants.java (hunk 6 above)
Comment (Yasara123) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/util/HttpConstants.java]: Fixed in c117e5f

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java (hunk 4 above)
Comment (Yasara123) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: Fixed in  	074b82b

**Pull request merged**